docs/nlu/components.rst

```diff
@@ -82,15 +82,30 @@ SpacyNLP
           # between these two words, therefore setting this to `true`.
           case_sensitive: false
 
-Featurizers
------------
+Text Featurizers
+----------------
+
+Featurizers are divided into two different categories: sparse featurizers and dense featurizers.
+Sparse featurizers are featurizers that return feature vectors with a lot of missing values, e.g. zeros.
+As those feature vectors would normally take up a lot of memory, we store them as sparse features.
+Sparse features only store the values that are non zero and their positions in the vector.
+Thus, we save a lot of memroy and are able to train on larger dataset.
+
+By default all featurizers will return a matrix of length (1 x feature-dimension).
```

Comment:
does not explain **why** I would use this

Reply:
currently, you have no choice 😅

Reply:
Yeah, it is not an option, it is a fact. Or do you mean why you would use ``return_sequence`` equal to True?

Reply:
yes this was commenting about this paragraph (sorry, should have been clear): why would a user want to the change value of  `return_sequence`? (if there is no reason for them, also fine but should be stated as well)